src/lib/EDSL.mli

```diff
@@ -92,6 +92,35 @@ val string : string -> string t
 val int : int -> int t
 val bool : bool -> bool t
 
+module Integer : sig
+  val to_string : int t -> string t
+  val of_string : string t -> int t
+  val bin_op : int t -> [ `Div | `Minus | `Mult | `Plus | `Mod ] -> int t -> int t
+  val add : int t -> int t -> int t
```

Comment:
Sorry for the misplacement, this is in reference to `bin_op`.

My usual preference for regular variants in specifying fully formed api's. Can't really imagine any more standard operators that you can define. But you know, just a style quibble, feel free to disregard.